lib/Models/CswCatalogGroup.js

```diff
@@ -447,13 +481,29 @@ Otherwise, if reloading doesn\'t fix it, please report the problem by sending an
 
 };
 
+function createWpsItemFromUri(terria, uri, title) {
+    var wps = new WebProcessingServiceCatalogGroup(terria);
+    wps.name = title;
+    wps.url = uri;
+
+    return wps.load().then(function() {
+        return wps;
+    });
+}
+
+function addProcessesToGroup(wpsPromise, wpsGroup) {
+    wpsPromise.then(function(wpsItem) {
```

Comment:
The best way to handle this case is to make sure that all promises are returned from `_load`.  Then other parts of TerriaJS will automatically handle errors and display them to the user.